rfcs/2021-11-22-design-tokens.md

```diff
@@ -0,0 +1,108 @@
+3. **Interoperability** - Design tokens allow a huge amount of flexibility for moving a design system to a new platform. Should an iOS/Android app ever be built, a new web application created, etc.. That system would have a framework ready to utilize for its theming which would be consistent with the other brand applications.
+4. **Opportunity for theming** - Decoupling design rules from the codebase allowing the transposition-ing of these static design decisions in various ways, i.e. theming/high contrast. By replacing all tokens in a system with new tokens you can essentially theme swap, assuming the consuming system is utilizing tokens at all levels appropriately.
+
+## Risks
+
+1. There is a small risk that implementing tokens could break many things at once, as many design rules would be referencing these across an application. With tools like storybook, visual regression and QA this would be a part of every applications standard testing procedure and should be easy to resolve.
+2. There can be a reductive quality to design tokens in that there might be a more limited set of colors/spacing options/etc.. to work with for the consuming applications. Someone might have to sell consuming systems on the advantages to brand consistency, which they hopefully already understand.
+
+## Scope/Outline
+
+**An outline for how this process could proceed:**
+
+1. The SSOT should be decided on, at the moment I see two options:
+
+   1. **The DSM becomes the source of truth**. Designers and developers work together to consolidate the current set of design rules residing in the codebase/Sketch and place them into the DSM. In this case, with that work done, an application can be written in the design system to utilize the [DSM Token API](https://support.invisionapp.com/hc/en-us/articles/360049950931) to periodically update it's set of scss variables with data from these design tokens and store them in a new NPM package.
+   2. **Tokens are modified directly as sets of JSON files**. Designers and developers work together to consolidate tokens in a hierarchy which makes the most sense for all platforms. These tokens could be stored in a new NPM package. Changes to tokens would require a pull request and have a release cadence.
+
+2. Once the SSOT is determined, tokens should be created in the appropriate format and at minimum, populated to form a set required to serve the current consuming applications. Ongoing work could be ticketed to fill out the entire token library. Items which should be tokenized can be decided on by the teams. These first tokens would represent the most basic items such as spacing, color, type, breakpoints, animation, border.. i.e..
```

Comment:
```suggestion
2. Once the SSOT is determined, tokens should be created in the appropriate format. At minimum tokens should also be populated to form a set required to serve the current consuming applications. Ongoing work could then be ticketed to fill out the entire token library. Items which should be tokenized can be decided on by the teams. These first tokens would represent the most basic items such as spacing and color values, font families and type sizes, breakpoints, animation easing values, border thickness, etc. An example of a color token setup could be:
```

Minor tweaks to make the example basic tokens more discrete.

Reply:
>Items which should be tokenized can be decided on by the teams

Should we include this? It suggests to me that different teams can create tokens in a vacuum, but I'm assuming we as the Experience team all collectively decide on what tokens get shared with consuming sources.

Reply:
> Items which should be tokenized can be decided on by the teams.

I had the same thought here as Eric. I don't think product teams should be deciding on what gets tokenized - I think product teams should help inform what should be tokenized through what is needed in terms of flexibility

Reply:
Oh that's just poor phrasing, I'll update that.. I meant between design and dev on our team.. But i'll likely just remove that line. 👍